C compilation error depending on the syntax with fixed size arrays

### Describe the bug

Adding brackets around a fixed sized array in a return results in a C compilation failure, with the noticeable output " error: ';' expected (got "}") ".

### Reproduction Steps
```v
fn main() {

    varraynob:=arraynob()
    println(varraynob)

    varrayb:=arrayb()
    println(varrayb)

}

//Works
fn arraynob() [3]int {
    return [1,2,3]! //No brackets
}

//Doesn't work!
fn arrayb() [3]int {
    return([1,2,3]!) //Brackets
}
```
### Expected Behavior

The brackets of the same color as the "return" make it easier to see what's returned, at least for me, and it should not behave differently if they are here or not.

### Current Behavior
```
Complete error log (with the code snippet provided):
======== Output of the C Compiler ("path"\v\thirdparty\tcc\tcc.exe) ========
"path"/Temp/v_0/snippet.01K8TV9HHW1B2JACG4CSG43M3M.tmp.c:387: warning: WINVER redefined

In file included from "path"/Temp/v_0/snippet.01K8TV9HHW1B2JACG4CSG43M3M.tmp.c:511:

"path"/v/thirdparty/tcc/include/winapi/synchapi.h:13: warning: CreateEvent redefined

In file included from "path"/Temp/v_0/snippet.01K8TV9HHW1B2JACG4CSG43M3M.tmp.c:511:

"path"/v/thirdparty/tcc/include/winapi/synchapi.h:117: warning: OpenEvent redefined

In file included from "path"/Temp/v_0/snippet.01K8TV9HHW1B2JACG4CSG43M3M.tmp.c:511:

"path"/v/thirdparty/tcc/include/winapi/synchapi.h:167: warning: CreateMutex redefined

"path"/Temp/v_0/snippet.01K8TV9HHW1B2JACG4CSG43M3M.tmp.c:5313: error: ';' expected (got "}")
==========================================================================================
======== Output of the C Compiler (gcc) ========
"path"\Temp\v_0\snippet.01K8TV9HHW1B2JACG4CSG43M3M.tmp.c: In function 'main__arrayb':
"path"\Temp\v_0\snippet.01K8TV9HHW1B2JACG4CSG43M3M.tmp.c:5313:38: error: expected ';' before '}' token
 5313 |         memcpy(_t1.ret_arr, ({1, 2, 3}), sizeof(Array_fixed_int_3));
      |                                      ^
      |                                      ;
"path"\Temp\v_0\snippet.01K8TV9HHW1B2JACG4CSG43M3M.tmp.c:5313:35: warning: passing argument 2 of 'memcpy' makes pointer from integer without a cast [-Wint-conversion]
 5313 |         memcpy(_t1.ret_arr, ({1, 2, 3}), sizeof(Array_fixed_int_3));
      |                             ~~~~~~^~~~~
      |                                   |
      |                                   int
In file included from "path"\Temp\v_0\snippet.01K8TV9HHW1B2JACG4CSG43M3M.tmp.c:327:
"path"/winlibs-x86_64-posix-seh-gcc-15.1.0-mingw-w64ucrt-13.0.0-r2/mingw64/x86_64-w64-mingw32/include/string.h:50:76: note: expected 'const void * restrict' but argument is of type 'int'
   50 |   void * __cdecl memcpy(void * __restrict__ _Dst,const void * __restrict__ _Src,size_t _Size) __MINGW_ATTRIB_DEPRECATED_SEC_WARN;
      |                                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~
================================================
```
### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.12 676fb62 

### Environment details (OS name and version, etc.)

Windows 11 25H2
Compiled w/ v "filename" -show-c-output and the included tcc
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/builtin/array_test.v
+++ b/vlib/builtin/array_test.v
@@ -806,6 +806,11 @@ fn test_fixed_array_literal_eq() {
 
 	assert [[1, 1]!, [2, 2]!] == [[1, 1]!, [2, 2]!]
 	assert [[1, 1]!, [2, 2]!] != [[1, 2]!, [2, 3]!]
+
+	// vfmt off
+	assert ([1, 2, 3]!) == [1, 2, 3]!
+	assert (([1, 2, 3]!)) == [1, 2, 3]!
+	// vfmt on
 }
 
 fn test_sort() {

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -474,6 +474,13 @@ fn (mut g Gen) assign_stmt(node_ ast.AssignStmt) {
 				is_fixed_array_init = val.is_fixed
 				has_val = val.has_val
 			}
+			ast.ParExpr {
+				if val.expr is ast.ArrayInit {
+					array_init := val.expr as ast.ArrayInit
+					is_fixed_array_init = array_init.is_fixed
+					has_val = array_init.has_val
+				}
+			}
 			ast.CallExpr {
 				is_call = true
 				if val.comptime_ret_val {
@@ -1025,6 +1032,19 @@ fn (mut g Gen) assign_stmt(node_ ast.AssignStmt) {
 								g.write(', sizeof(${styp}))')
 							} else {
 								g.array_init(val, cvar_name)
+							}
+						} else if val is ast.ParExpr && val.expr is ast.ArrayInit {
+							array_init := val.expr as ast.ArrayInit
+							cvar_name := c_name(ident.name)
+							if array_init.is_fixed && ident.name in g.defer_vars {
+								g.go_before_last_stmt()
+								g.empty_line = true
+								g.write('memcpy(${cvar_name}, ')
+								g.write('(${styp})')
+								g.array_init(array_init, cvar_name)
+								g.write(', sizeof(${styp}))')
+							} else {
+								g.array_init(array_init, cvar_name)
 							}
 						} else if val_type.has_flag(.shared_f) {
 							g.expr_with_cast(val, val_type, var_type)

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -3895,7 +3895,12 @@ fn (mut g Gen) expr(node_ ast.Expr) {
 		}
 		ast.ParExpr {
 			g.write('(')
-			g.expr(node.expr)
+			if node.expr is ast.ArrayInit && node.expr.is_fixed && !g.inside_assign {
+				array_init := node.expr as ast.ArrayInit
+				g.fixed_array_init_with_cast(array_init, array_init.typ)
+			} else {
+				g.expr(node.expr)
+			}
 			g.write(')')
 		}
 		ast.PostfixExpr {

